src/main/java/de/tum/in/www1/artemis/service/connectors/localvc/LocalVCServletService.java

```diff
@@ -235,37 +245,64 @@ public void authenticateAndAuthorizeGitRequest(HttpServletRequest request, Repos
         log.debug("Authorizing user {} for repository {} took {}", user.getLogin(), localVCRepositoryUri, TimeLogUtil.formatDurationFrom(timeNanoStart));
     }
 
-    private User authenticateUser(String authorizationHeader) throws LocalVCAuthException {
+    private User authenticateUser(String authorizationHeader, ProgrammingExercise exercise, LocalVCRepositoryUri localVCRepositoryUri) throws LocalVCAuthException {
 
         UsernameAndPassword usernameAndPassword = extractUsernameAndPassword(authorizationHeader);
 
         String username = usernameAndPassword.username();
         String password = usernameAndPassword.password();
-
-        var user = userRepository.findOneByLogin(username);
+        User user = userRepository.findOneByLogin(username).orElseThrow(LocalVCAuthException::new);
 
         try {
             SecurityUtils.checkUsernameAndPasswordValidity(username, password);
+        }
+        catch (AccessForbiddenException | AuthenticationException e) {
+            if (!StringUtils.isEmpty(password)) {
+                log.warn("Failed login attempt for user {} with password {} due to issue: {}", username, password, e.getMessage());
+            }
+            throw new LocalVCAuthException(e.getMessage());
```

Comment:
Can you fix this violation?

```
  java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'no classes should depend on classes that have simple name 'StringUtils' and not reside in any package ['org.apache.commons.lang3', 'org.springframework.util']' was violated (1 times):
  Method <de.tum.in.www1.artemis.service.connectors.localvc.LocalVCServletService.authenticateUser(java.lang.String, de.tum.in.www1.artemis.domain.ProgrammingExercise, de.tum.in.www1.artemis.service.connectors.localvc.LocalVCRepositoryUri)> calls method <net.sourceforge.plantuml.StringUtils.isEmpty(java.lang.CharSequence)> in (LocalVCServletService.java:260)
```

Reply:
Fixed it by using a different import for `StringUtils` -> https://github.com/ls1intum/Artemis/pull/8929/commits/b9e8b6d4a274e57bf13576d1ea84d978bc88c002